addons/html_editor/static/src/link/link.js

```diff
@@ -0,0 +1,33 @@
+import { Component, useRef, useState } from "@odoo/owl";
+import { useOverlay } from "../core/overlay_plugin";
+
+export class LinkPopover extends Component {
+    static template = "html_editor.linkPopover";
+    static props = {
+        // overlay props
+        dispatch: Function,
+        offset: Function,
+        el: {
+            validate: (el) => el.nodeType === Node.ELEMENT_NODE,
+        },
+        // link props
+        linkState: Object,
+    };
+    setup() {
+        this.urlInput = useRef("urlInput");
+        this.state = useState({
+            editing: this.props.linkState.linkElement.href ? false : true,
+            url: this.props.linkState.linkElement.href || "",
+        });
```

Comment:
Why not use directly the linkState ? With this design, if the plugin edit linkElement.href, you will not be notify

Reply:
Or you dont need to put linkState in a reactive

Reply:
@FrancoisGe Thanks for checking! I'm not an expert using the reactive, I got there cause I had a discussion with Nicolas about how to modify a link element in props in selection change. It's probably not the best way to use the reactive because I wanted to only make the link element changeable.

Just make sure I understood correctly, so it would be better we use the linkState like below? :pray: 

```js
//initialize the linkState
this.linkState = reactive({
            linkElement: null,
            editing: false,
            url: "",
        });
// update linkState when selection change
this.linkState.linkElement = linkel;
this.linkState.editing = linkel.href ? false : true;
this.linkState.url = linkel.href || "";

// we use the linkState in the popover component like this
this.state =  useState(this.props.linkState)
```